### PR TITLE
Upgrade components

### PR DESCRIPTION
**Pull request by adriananeci**
Notable changes:
- k8s from 1.21 to 1.23
- vagrant image from ubuntu/bionic64 to  ubuntu/focal64
- nginx ingress controller from 0.43.0 to 1.1.0
- cilium from 1.10 to 1.11

Network CNIs, except cilium, weren't upgraded(not sure if they are still used).

Validated on vagrant + virtualbox.
```
root@ubuntu-focal:~# k get no -owide
NAME            STATUS   ROLES                  AGE     VERSION   INTERNAL-IP       EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION     CONTAINER-RUNTIME
control-plane   Ready    control-plane,master   7m13s   v1.23.1   192.168.234.100   <none>        Ubuntu 20.04.3 LTS   5.4.0-91-generic   docker://20.10.12
node01          Ready    <none>                 3m53s   v1.23.1   192.168.234.101   <none>        Ubuntu 20.04.3 LTS   5.4.0-91-generic   docker://20.10.12
node02          Ready    <none>                 37s     v1.23.1   192.168.234.102   <none>        Ubuntu 20.04.3 LTS   5.4.0-91-generic   docker://20.10.12
root@ubuntu-focal:~# k get po -A
NAMESPACE       NAME                                         READY   STATUS      RESTARTS   AGE
ingress-nginx   ingress-nginx-admission-create-8r2tq         0/1     Completed   0          7m2s
ingress-nginx   ingress-nginx-admission-patch-v6p5f          0/1     Completed   0          7m2s
ingress-nginx   ingress-nginx-controller-84b9776df8-74k5t    1/1     Running     0          7m1s
kube-system     cilium-ldm8n                                 1/1     Running     0          44s
kube-system     cilium-operator-78cc86cdcf-g5jjt             1/1     Running     0          7m1s
kube-system     cilium-operator-78cc86cdcf-scwmq             1/1     Running     0          7m1s
kube-system     cilium-qx4zb                                 1/1     Running     0          4m
kube-system     cilium-w8mzs                                 1/1     Running     0          7m2s
kube-system     coredns-64897985d-fpjwr                      1/1     Running     0          7m1s
kube-system     dashboard-metrics-scraper-86bf6486bc-lsn2s   1/1     Running     0          7m1s
kube-system     etcd-control-plane                           1/1     Running     0          7m15s
kube-system     hubble-relay-56d45f97f8-qdmrj                1/1     Running     0          7m1s
kube-system     hubble-ui-54fb86b9f4-7nfvw                   3/3     Running     0          7m1s
kube-system     kube-apiserver-control-plane                 1/1     Running     0          7m18s
kube-system     kube-controller-manager-control-plane        1/1     Running     0          7m18s
kube-system     kube-proxy-bvdm2                             1/1     Running     0          7m2s
kube-system     kube-proxy-j7tjm                             1/1     Running     0          4m
kube-system     kube-proxy-vw9wp                             1/1     Running     0          44s
kube-system     kube-scheduler-control-plane                 1/1     Running     0          7m19s
kube-system     kubernetes-dashboard-56647955b9-bmhvf        1/1     Running     0          7m1s
kube-system     metrics-server-d9c898cdf-pqs8x               1/1     Running     0          7m1s
```

Once merged, the next steps will be to validate it in AWS and adjust labs specs accordingly.

### CODE REVIEW

File: src/scripts/control-plane.sh [hunk 1]
```diff
@@ -45,7 +44,13 @@ kubectl apply -f /src/manifests/metrics-server/
 kubectl -n kube-system scale deployment coredns --replicas=1
 
 # get admin token
-kubectl describe secret "$(kubectl get secrets | grep cluster | cut -d ' ' -f1)" | grep token:  | tr -s ' ' | cut -d ' ' -f2 > /src/output/cluster-admin-token
+until kubectl get secrets -o name | grep cluster
+do
+    echo "Waiting for admin secret to be created"
```
Comment (victorvarza) [inline src/scripts/control-plane.sh:49]: isn't creted at line 12 during `kubeadm init`?

File: src/utils/vagrant-template/setup.sh [hunk 2]
```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env bash
```
Comment (victorvarza) [inline src/utils/vagrant-template/setup.sh:1]: I think vagrant-template is not used anymore, we can delete it.

File: src/scripts/control-plane.sh (hunk 1 above)
Comment (adriananeci) [reply src/scripts/control-plane.sh:49]: Yes, but it is taking a while until it isavailable in kubectl output, hence this while loop.

File: src/utils/vagrant-template/setup.sh (hunk 2 above)
Comment (adriananeci) [reply src/utils/vagrant-template/setup.sh:1]: I would prefer to do it in a follouwp PR to avoid mixing up too many things.

**Review by victorvarza — APPROVED** [review]

**Pull request merged**